tests/unit/cli/test_import_module_proxy.py

```diff
@@ -0,0 +1,30 @@
+import importlib
+from unittest import TestCase
+
+from parameterized import parameterized
+
+from samcli.cli import hidden_imports
+from samcli.cli.import_module_proxy import attach_import_module_proxy, MissingDynamicImportError
+
+
+class TestImportModuleProxy(TestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        attach_import_module_proxy()
```

Comment:
Do we need to have a "detach" and call it in teardown in order to ensure that subsequent unit tests are not importing through the proxy?  I'm not sure this is needed, and I'm also not sure whether unit tests are run in parallel by the same python interpreter which would also cause this concern.  The concern is that the proxy may interfere with tests that need to load classes not under `test`.

Reply:
Thanks for suggestions @marekaiv, I've added a detach method to remove the proxy after the unit tests are completed. I've also tested running unit tests in parallel but I haven't see any other test failing for this reason. But since this is really depend on a race condition and might happen in the future if we choose to run unit tests in parallel, I've added a comment for the test class, so that there could be some clues for investigation.